pkgs/create-neon/src/cache/npm.ts

```diff
@@ -1,11 +1,16 @@
 import { Cache } from "../cache.js";
 
 export class NPM implements Cache {
-  readonly org: string | null;
+  readonly org: string;
 
   readonly type: string = "npm";
 
-  constructor(org: string | null) {
-    this.org = org;
+  constructor(pkg: string, org?: string) {
+    this.org = org || NPM.inferOrg(pkg);
+  }
+
+  static inferOrg(pkg: string): string {
+    const m = pkg.match(/^@([^/]+)\/([^/]+)/);
+    return "@" + (m ? m[1] : pkg);
```

Comment:
FYI, with fancy JS features (optional chaining and null-ish coalescing).

```js
return `@${m?.[1] ?? pkg}`;
```

Reply:
ooh much nicer